sourcegraph/src-cli
# [language-platform] update src-cli to handle scip file upload

Fixes https://github.com/sourcegraph/sourcegraph/issues/45078

Before this PR src codeintel upload would translate a SCIP file to LSIF prior to upload. With this PR it now instead checks with the instance to see if it can handle SCIP uploads. This is done with a HEAD request to the upload endpoint it then updates the headers to reflect the SCIP metadata.

### Test plan



To test properly you will need the changes from this PR https://github.com/sourcegraph/sourcegraph/pull/45269

1. Update `go.mod` in this repo to point to your local environment with the changes above.
Example:

```yaml
replace github.com/sourcegraph/sourcegraph/lib => /Users/<YOUR NAME>/go/src/sourcegraph/lib
```
2. Build src-cli to have the current build. i.e `go build` inside the src-cli/cmd/src directory
3. Navigate to any typescript repo and use https://github.com/sourcegraph/scip-typescript to create an index to upload. I tend to index `sourcegraph/sourcegraph/client/web`
4. Use your newly built src-cli (from step 2) to upload your newly created index (from step 3)
```terminal
➜ SRC_ENDPOINT=https://sourcegraph.test:3443 ../../../src-cli/cmd/src/src lsif upload -indexer=scip-typescript -indexerVersion=v0.3.3
```
5. Success you can hit the new endpoint.

At the moment, the instance cannot handle SCIP. But this PR (https://github.com/sourcegraph/sourcegraph/pull/45104) should handle that part.
